### PR TITLE
vcr: GH presubmit job

### PR DESCRIPTION
**Pull request by gemmahou**
### Change description

Setup GH presubmit job for vcr.

Test against #1552 

Note:
The execution time per test depends on different resources, and their log size. In [screenshot](https://screenshot.googleplex.com/938oupPE63rBwMD.png) we can see resource `globalcomputeforwardingrule` takes ~211s to finish, checking the [vcr log](https://github.com/gemmahou/k8s-config-connector/blob/1fff296b7412de52e83985ac2146bd91d0a99981/tests/e2e/testdata/vcr-cassette/TestAllInSeries_fixtures_fullalloydbcluster-tf.yaml#L2191), it has 42 interactions to be replay.

### Tests you have done

- [x] Run `make ready-pr` to ensure this PR is ready for review.
- [x] Perform necessary E2E testing for changed resources.

### CODE REVIEW

**Comment by justinsb** [comment]
A few nits, but not blocking.  Will add hold to make sure you see comments (and I wasn't sure if you said you were worried about the time it takes?), feel free to remove hold.

/approve
/lgtm
/hold

File: config/tests/samples/create/harness.go [hunk 1]
```diff
@@ -348,28 +350,34 @@ func NewHarnessWithOptions(ctx context.Context, t *testing.T, opts *HarnessOptio
...
-
-		if kccConfig.HTTPClient == nil {
-			httpClient, err := google.DefaultClient(ctx, gcp.ClientScopes...)
-			if err != nil {
-				t.Fatalf("error creating the http client to be used by DCL: %v", err)
-			}
-			kccConfig.HTTPClient = httpClient
-		}
+		// In replay mode, RealTransport is unnecessary because we simply replay existing cassettes.
+		// We only run replay mode in the open-source repo, where we are unable to create a real DCL HTTP client.
```
[the remark this answers is not in the document]
Comment (gemmahou) [reply config/tests/samples/create/harness.go]: Removed unnecessary comments.

File: config/tests/samples/create/harness.go [hunk 2]
```diff
@@ -348,28 +350,34 @@ func NewHarnessWithOptions(ctx context.Context, t *testing.T, opts *HarnessOptio
...
 		opts := &recorder.Options{
-			CassetteName:  filepath.Join(path, "dcl"),
-			Mode:          vcrMode,
-			RealTransport: kccConfig.HTTPClient.Transport,
+			CassetteName: filepath.Join(path, "dcl"),
+			Mode:         vcrMode,
+		}
+		// In record mode, use the real DCL HTTP client's transport as the recorder's transport.
```
[the remark this answers is not in the document]
Comment (gemmahou) [reply config/tests/samples/create/harness.go]: Updated_

File: config/tests/samples/create/harness.go [hunk 3]
```diff
@@ -608,6 +616,52 @@ func MaybeSkip(t *testing.T, name string, resources []*unstructured.Unstructured
 			}
 		}
 	}
+	if os.Getenv("E2E_GCP_TARGET") == "vcr" {
+		switch name {
```
[the remark this answers is not in the document]
Comment (gemmahou) [reply config/tests/samples/create/harness.go:619]: I'm using tests in presubmit constants([code](https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/pkg/test/constants/presubmitconstants.go#L31)) to determine presubmit resources, one test from each service. 
I'll create follow-up PRs to generate VCR records for each service, so I added a TODO here to manage this list by service once we have more VCR coverage per service level.

**Comment by gemmahou** [comment]
> A few nits, but not blocking. Will add hold to make sure you see comments (and I wasn't sure if you said you were worried about the time it takes?), feel free to remove hold.
> 
> /approve /lgtm /hold

Yeah looking at the presubmit jobs([screenshot](https://screenshot.googleplex.com/8DMQtuxjpibiVaZ.png)), vcr fixtures test takes 42min to finish and mock takes 20min. Also vcr only covers about 40 presubmit tests now, while mock covers more.

**Comment by gemmahou** [comment]
#1623 will speed up the vcr test. 
Will need to update existing cassettes as well. Keep this PR on hold.

**Pull request merged**